x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowIndexAction.java

```diff
@@ -297,6 +293,19 @@ static void validate(
         followerMapperService.merge(leaderIndex, MapperService.MergeReason.MAPPING_RECOVERY);
     }
 
+    static Tuple<String, String> splitClusterAliasAndIndexName(Set<String> remoteClusterNames, String leaderIndexExpression) {
+        int indexOf = leaderIndexExpression.indexOf(RemoteClusterService.REMOTE_CLUSTER_INDEX_SEPARATOR);
+        if (indexOf != -1) {
+            String clusterAlias = leaderIndexExpression.substring(0, indexOf);
+            if (remoteClusterNames.contains(clusterAlias) == false) {
+                throw new IllegalArgumentException("unknown cluster alias [" + clusterAlias + "]");
```

Comment:
don't you need to handle the case where you have a local index that contains ":" ?

Reply:
Yes, but that will be addressed in a different PR. This change is just about validating whether the cluster alias exists and whether it has been configured.

That validation can be added to `FollowIndexAction.Request.validate()` method. If a leaderIndex contains more than 1 `:` then we can fail the request.

Reply:
but is the plan to go local if no remote cluster is found and there is an index with that name (although it contains ":") ? Seems like we may end up duplicating the bw comp logic from RemoteClusterService, and we already duplicate the splitting logic. I am fine with changing this later but I am not 100% convinced that this new method belongs here.